akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -287,15 +287,9 @@ private[akka] class RemoteWatcher(
    * does not exist.
    */
   def reWatch(address: Address): Unit =
```

Comment:
To my understandig, I think it is risky to trigger DeathWatchNotification when seeing new uid. There can be some new watches established with actors on system with the new uid. There is no way to distinguish them unless we store the uid along with all the watches (the tuples of watcher, watchee)
But I am more likely to miss something than you ;)